numpy/_core/src/multiarray/calculation.c

```diff
@@ -637,8 +637,7 @@ PyArray_Round(PyArrayObject *a, int decimals, PyArrayObject *out)
                 return (PyObject *)out;
             }
             else {
-                Py_INCREF(a);
-                return (PyObject *)a;
+                return PyArray_Copy(a);
```

Comment:
```suggestion
                return PyArray_NewCopy(a, NPY_KEEPORDER);
```

Reply:
Oh... This made the test fail, but I hope that needs a test adjustment only :).

Reply:
Well, I parroted a `PyArray_Copy` from the complex dtype code path of this same function, so if we insist on `NPY_KEEPORDER` for real dtypes, we need to follow for complex dtypes, too. Done now.